HealthMonitoring.Management.Core/EndpointStatsManager.cs

```diff
@@ -1,24 +1,27 @@
 ﻿using System;
-using System.Collections.Concurrent;
+using System.Linq;
 using System.Threading;
 using Common.Logging;
 using HealthMonitoring.Configuration;
 using HealthMonitoring.Management.Core.Repositories;
 using HealthMonitoring.Model;
+using HealthMonitoring.Queueing;
 using HealthMonitoring.TimeManagement;
 
 namespace HealthMonitoring.Management.Core
 {
     public class EndpointStatsManager : IEndpointStatsManager, IDisposable
     {
-        private static readonly ILog _logger = LogManager.GetLogger<EndpointStatsManager>();
+        private const int MaxStatsQueueSize = 10000;
+        private static readonly ILog Logger = LogManager.GetLogger<EndpointStatsManager>();
         private readonly IEndpointStatsRepository _repository;
         private readonly IMonitorSettings _settings;
         private readonly ITimeCoordinator _timeCoordinator;
         private readonly Thread _cleanerThread;
         private readonly Thread _writerThread;
         private readonly IEndpointMetricsForwarderCoordinator _metricsForwarderCoordinator;
-        private readonly BlockingCollection<Tuple<EndpointIdentity, EndpointMetadata, EndpointHealth>> _statsQueue = new BlockingCollection<Tuple<EndpointIdentity, EndpointMetadata, EndpointHealth>>();
+        private readonly OutgoingQueue<Tuple<EndpointIdentity, EndpointMetadata, EndpointHealth>> _statsQueue = new OutgoingQueue<Tuple<EndpointIdentity, EndpointMetadata, EndpointHealth>>(MaxStatsQueueSize);
```

Comment:
Limit the queue size to 10000 items. If limit reached, the new items will overwrite the oldest ones, but memory usage will not grow.